src/payload.rs

```diff
@@ -36,12 +36,34 @@ use {
 pub struct RpcSendTransactionConfigWithBlockList {
     #[serde(flatten)]
     pub config: Option<RpcSendTransactionConfig>,
-    pub blocklist_pdas: Vec<Pubkey>,
+    /// base58-encoded pubkeys
+    pub blocklist_pdas: Option<Vec<String>>,
 }
 
-impl From<RpcSendTransactionConfigWithBlockList> for RpcSendTransactionConfig {
-    fn from(extended: RpcSendTransactionConfigWithBlockList) -> Self {
-        extended.config.unwrap_or_default()
+impl RpcSendTransactionConfigWithBlockList {
+    /// Converts the optional string-based blocklist_pdas into a Vec<Pubkey>
+    /// Invalid pubkeys are filtered out
+    pub fn blocklist_pubkeys(&self) -> Vec<solana_sdk::pubkey::Pubkey> {
+        match &self.blocklist_pdas {
+            Some(keys) => keys
+                .iter()
+                .filter_map(
+                    |key_str| match solana_sdk::pubkey::Pubkey::from_str(key_str) {
+                        Ok(pubkey) => Some(pubkey),
+                        Err(_) => None,
+                    },
+                )
+                .collect(),
+            None => Vec::new(),
+        }
+    }
+
+    /// Get a safe reference to config or a default config if None
+    pub fn get_config(&self) -> RpcSendTransactionConfig {
+        match &self.config {
+            Some(config) => *config,
+            None => RpcSendTransactionConfig::default(),
+        }
```

Comment:
Does unwrap_or work in this case?

` self.config.unwrap_or(RpcSendTransactionConfig::default())`